Code/Editor/Lib/Tests/test_EditorCamera.cpp

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AZTestShared/Math/MathTestHelpers.h>
+#include <AtomToolsFramework/Viewport/ModularViewportCameraController.h>
+#include <AzCore/UnitTest/TestTypes.h>
+#include <AzFramework/Viewport/ViewportControllerList.h>
+#include <AzTest/GemTestEnvironment.h>
+#include <AzToolsFramework/API/EditorCameraBus.h>
+#include <AzToolsFramework/ToolsComponents/TransformComponent.h>
+#include <EditorModularViewportCameraComposer.h>
+
+namespace UnitTest
+{
+    class EditorCameraTestEnvironment : public AZ::Test::GemTestEnvironment
+    {
+        // AZ::Test::GemTestEnvironment overrides ...
+        void AddGemsAndComponents() override;
+    };
+
+    void EditorCameraTestEnvironment::AddGemsAndComponents()
+    {
+        AddDynamicModulePaths({ "Camera.Editor.Gem" });
```

Comment:
this is baking the name of the module, instead, we should pass it as a compilation variable to this file and take it from CMake. That way if the module name changes, this will always get the right name.

You can use `$<TARGET_FILE_BASE_NAME:Camera.Editor>` to get that

Reply:
Is there a more complete example of how to do this? I'm not sure I'm familiar with all the steps involved, are they any docs on this or could we briefly catch-up offline about it? Thanks!